google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -2398,6 +2402,160 @@ private void testBucketPolicyRequesterPays(boolean requesterPays) {
             bucketOptions));
   }
 
+  private void testBucketPolicyV3RequesterPays(boolean requesterPays) {
+    if (requesterPays) {
```

Comment:
Conditionals around asserts are a code smell. This should be two tests, one with and one without requesterPays.

Reply:
Or two helper methods